src/time.cpp

```diff
@@ -87,20 +87,27 @@ namespace g3 {
 
    int timespec_get(struct timespec* ts/*, int base*/) {
 #ifdef __MACH__
-      // std::timespec_get or posix clock_gettime)(...) are not
-      // implemented on OSX
-      // @return value of base if successful, else zero
-      struct timeval now = {};
-      int rv = gettimeofday(&now, nullptr);
-      if (-1 == rv) {
-         return rv;
-      }
-      // error mode. just return sec, microsecond
-      ts->tv_sec  = now.tv_sec;
-      ts->tv_nsec = now.tv_usec * 1000;
-      return 0;
+	   // std::timespec_get or posix clock_gettime)(...) are not
+	   // implemented on OSX
+	   // @return value of base if successful, else zero
+	   struct timeval now = {};
+	   int rv = gettimeofday(&now, nullptr);
+	   if (-1 == rv) {
+		   return rv;
+	   }
+	   // error mode. just return sec, microsecond
+	   ts->tv_sec = now.tv_sec;
+	   ts->tv_nsec = now.tv_usec * 1000;
+	   return 0;
 #elif(defined(WIN32) || defined(_WIN32) || defined(__WIN32__))
-	   return timespec_get(ts, TIME_UTC);
+	   static auto os =
```

Comment:
Hmm. I haven't seen this exercise before.  Can you explain what happens here. What exactly is the `os` that you get when you subtract `system clock time since epoch` with` high resolution clock time since epoc`?